R/rc_sim.R

```diff
@@ -185,77 +181,46 @@ rc_sim <- function(soil_properties,
-  if(initialize == TRUE){
-    
-    # set TOC to ton C / ha
-    dt.soc[, toc := toc * 0.001]
-    
-    # time correction (is 12 in documentation Chantals' study, not clear why, probably due to fixed time step calculation)
-    timecor = 1
-    
-    # set rate modifying parameters
-    abc <- rothc.parms$abc
-    
-    # CDPM pool (ton C / ha)
-    cdpm.ini <- rothc.event[var == 'CDPM',list(time,value)]
-    cdpm.ini[,cf_abc := abc(time)]
-    cdpm.ini <- cdpm.ini[,((sum(value) * 0.001 / max(time)) / (mean(cf_abc)/timecor))/k1]
-    dt.soc[, cdpm.ini := mean(cdpm.ini)]
-    
-    # CRPM pool (ton C / ha)
-    crpm.ini = rothc.event[var == 'CRPM',list(time,value)]
-    crpm.ini[,cf_abc := abc(time)]
-    crpm.ini <- crpm.ini[,((sum(value) * 0.001 / max(time)) / (mean(cf_abc)/timecor))/k2]
-    dt.soc[, crpm.ini := mean(crpm.ini)]
-    
-    # CIOM pool (ton C / ha)
-    dt.soc[, ciom.ini := 0.049 * toc^1.139]
-    
-    # CBIOHUM pool (ton C /ha)
-    dt.soc[,biohum.ini := toc - ciom.ini - crpm.ini - cdpm.ini]
-    
-    # set to defaults when RPM and DPM inputs exceeds 70% / 50% of total C to avoid negative values for initial C pools
-    dt.soc[biohum.ini <0, cdpm.ini := 0.015 * (toc-ciom.ini)]
-    dt.soc[biohum.ini <0, crpm.ini := 0.125 * (toc-ciom.ini)]
-    dt.soc[, biohum.ini := toc-ciom.ini - crpm.ini - cdpm.ini]
-    
-    # CBIO and CHUM pool
-    dt.soc[,cbio.ini := biohum.ini / (1 + k3 / k4)]
-    dt.soc[,chum.ini := biohum.ini / (1 + k4 / k3)]
+
+  # initialize the RothC pools if requested 
+  if(initialization_method != 'none'){ 
     
-    # Set the intial C pools (kg C / ha)
-    dt.soc[,CIOM0 := ciom.ini * 1000]
-    dt.soc[,CDPM0 := cdpm.ini * 1000]
-    dt.soc[,CRPM0 := crpm.ini * 1000]
-    dt.soc[,CBIO0 := cbio.ini * 1000]
-    dt.soc[,CHUM0 := chum.ini * 1000]
-  } else {
-    # Calculate carbon pools based on provided or default distribution (kg C / ha)
-    dt.soc[,CIOM0 := c_fractions$fr_IOM * ((toc*0.001)^1.139) * 1000]
-    dt.soc[,CDPM0 := c_fractions$fr_DPM * (toc-CIOM0)]
-    dt.soc[,CRPM0 := c_fractions$fr_RPM * (toc-CIOM0)]
-    dt.soc[,CBIO0 := c_fractions$fr_BIO * (toc-CIOM0)]
-    dt.soc[,CHUM0 := toc-CIOM0-CDPM0-CRPM0-CBIO0]
+    # derive the initial distribution of C pools (original data.tables are used as input)
+    c_fractions = as.list(rc_initialise(crops = rothc_rotation, 
+                                        amendment = rothc_amendment,
+                                        dt.time = dt.time,
+                                        dt.soc = dt.soc,
+                                        soil_properties = soil_properties,
+                                        rothc.event = rothc.event,
+                                        start_date = start_date,
+                                        dt.weather = weather,
+                                        rothc.parms = rothc.parms,
+                                        initialization_method = initialization_method))
     
-  }
   
+  } 
```

Comment:
there is no else command? should you have c_fractions <- rothc_parms$c_fractions?

Reply:
okido